advisors/spring-ai-advisors-vector-store/src/main/java/org/springframework/ai/chat/client/advisor/vectorstore/VectorStoreChatMemoryAdvisor.java

```diff
@@ -48,14 +49,24 @@
  * @author Christian Tzolov
  * @author Thomas Vitale
  * @author Oganes Bozoyan
+ * @author Mark Pollack
  * @since 1.0.0
  */
-public class VectorStoreChatMemoryAdvisor extends AbstractChatMemoryAdvisor<VectorStore> {
+public class VectorStoreChatMemoryAdvisor implements BaseChatMemoryAdvisor {
+
+	private static final Logger logger = LoggerFactory.getLogger(VectorStoreChatMemoryAdvisor.class);
+
+	public static final String CHAT_MEMORY_RETRIEVE_SIZE_KEY = "chat_memory_response_size";
```

Comment:
We could consider renaming this to something more explicit about what it's used for

```suggestion
	public static final String TOP_K = "chat_memory_vector_store_top_k";
```

Reply:
yea, also maybe move it into the VectorStore interface so it is easier to discover.  Let's consider tomorrow.